### PR TITLE
feat: Support the usage of clang-tidy when cross compiling with GCC.

### PR DESCRIPTION
**Pull request by Project579**
Currently clang-tidy will fail to run if using GCC with a target different from the host, this modification allows us to extract the default target and sysroot from a GCC compiler in order to feed it to clang-tidy.

### CODE REVIEW

**Review by aminya — CHANGES_REQUESTED** [review]
It seems that the MinGW tests are failing on Windows.

**Comment by Project579** [comment]
The tests seem to be passing now except for an expired certificate but that's not related to this PR.

**Review by aminya — APPROVED** [review]
Thanks very much! 🎉

File: src/StaticAnalyzers.cmake [hunk 1]
```diff
@@ -50,6 +50,131 @@ macro(enable_cppcheck CPPCHECK_OPTIONS)
...
+      COMMAND_ERROR_IS_FATAL ANY
+    )
+    execute_process(
+      COMMAND "${CMAKE_CXX_COMPILER}" "-dumpmachine"
+      OUTPUT_VARIABLE CLANG_TIDY_CXX_FLAGS_COMPILER_TARGET
+      COMMAND_ERROR_IS_FATAL ANY
+    )
+
+    string(STRIP "${CLANG_TIDY_CXX_FLAGS_COMPILER_TARGET}" CLANG_TIDY_CXX_FLAGS_COMPILER_TARGET)
+    set(CLANG_TIDY_CXX_FLAGS_COMPILER_TARGET "--target=${CLANG_TIDY_CXX_FLAGS_COMPILER_TARGET}")
```
Comment (aminya) [inline src/StaticAnalyzers.cmake:93]: There is a DetectCompiler module. I am not sure if this part can be simplified using that. In any case, we can probably refactor it into that module can introduce some macros for compiler/target detection.

**Pull request merged**